common/src/KokkosKernels_Sorting.hpp

```diff
@@ -17,27 +17,21 @@
 #define _KOKKOSKERNELS_SORTING_HPP
 
 #include "Kokkos_Core.hpp"
+#include "Kokkos_Sort.hpp"
 #include "KokkosKernels_SimpleUtils.hpp"     //for kk_exclusive_parallel_prefix_sum
 #include "KokkosKernels_ExecSpaceUtils.hpp"  //for kk_is_gpu_exec_space
 #include <type_traits>
 
 namespace KokkosKernels {
 
-namespace Impl {
-template <typename Value>
-struct DefaultComparator {
-  KOKKOS_INLINE_FUNCTION bool operator()(const Value lhs, const Value rhs) const { return lhs < rhs; }
-};
-}  // namespace Impl
-
 // ----------------------------
 // General device-level sorting
 // ----------------------------
 
 // Bitonic sort: sorts v according to the comparator object's operator().
 // Default comparator is just operator< for v's element type.
 template <typename View, typename ExecSpace, typename Ordinal,
-          typename Comparator = Impl::DefaultComparator<typename View::value_type>>
+          typename Comparator = Kokkos::Experimental::Impl::StdAlgoLessThanBinaryPredicate<typename View::value_type>>
```

Comment:
I am a bit worried about using something from Kokkos' Impl namespace, @dalg24 will scold you!

Reply:
Please don't use that

Reply:
Darn! 😄 I just wanted to match what Kokkos NestedSort does and not redefine the ``<`` comparator. I'll put it back.